File: benchmarks/pb_except/mono.php
<?php

function g($i) {
    throw new RuntimeException("Except $i");
}

function f($n) {
    $i = 0;
    while($i < $n) {
        try {
            g($i);
            assert(False);
        } catch(RuntimeException $e) {
            assert($e->getMessage() == "Except $i");
        }
        $i ++;
    }
}

function run_iter($n) {
    f($n);
}

?>

File: benchmarks/pb_except/mono.py
def g(i):
    raise RuntimeError("Except %s" % i)

def f(n):
    i = 0
    while i < n:
        try:
            g(i)
            assert False
        except RuntimeError as e:
            assert e.args[0] == "Except %s" % i
        i += 1

def run_iter(n):
    f(n)
